go/client/cmd_wallet_handle_uri.go

```diff
@@ -0,0 +1,301 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/keybase/cli"
+	"github.com/keybase/client/go/libcmdline"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/stellar1"
+	"github.com/keybase/go-framed-msgpack-rpc/rpc"
+	"github.com/keybase/stellarnet"
+	context "golang.org/x/net/context"
+)
+
+type CmdWalletHandleURI struct {
+	libkb.Contextified
+	uri string
+}
+
+func newCmdWalletHandleURI(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Command {
+	cmd := &CmdWalletHandleURI{
+		Contextified: libkb.NewContextified(g),
+	}
+
+	return cli.Command{
+		Name:         "handle-uri",
+		Usage:        "Handle a 'web+stellar:' URI manually",
+		ArgumentHelp: "<uri>",
+		Action: func(c *cli.Context) {
+			cl.ChooseCommand(cmd, "handle-uri", c)
+		},
+	}
+}
+
+func (c *CmdWalletHandleURI) ParseArgv(ctx *cli.Context) error {
+	c.uri = ctx.Args().First()
+	if c.uri == "" {
+		return errors.New("uri required")
+	}
+
+	return nil
+}
+
+func (c *CmdWalletHandleURI) Run() (err error) {
+	defer transformStellarCLIError(&err)
+
+	cli, err := GetWalletClient(c.G())
+	if err != nil {
+		return err
+	}
+
+	protocols := []rpc.Protocol{
+		NewIdentifyUIProtocol(c.G()),
+	}
+	if err := RegisterProtocolsWithContext(protocols, c.G()); err != nil {
+		return err
+	}
+
+	ui := c.G().UI.GetTerminalUI()
+
+	ui.Printf(c.yellow("Validating URI...") + "\n")
+	arg := stellar1.ValidateStellarURILocalArg{
+		InputURI: c.uri,
+	}
+	v, err := cli.ValidateStellarURILocal(context.Background(), arg)
+	if err != nil {
+		return err
+	}
+
+	ui.Printf(c.green("URI validated.") + "\n\n")
+	ui.Printf("Keybase validated the URI you submitted.  It was signed by\n\n")
+	ui.Printf("\t%s\n\n", ColorString(c.G(), "underline", v.OriginDomain))
+	if v.Message != "" {
+		ui.Printf("Message: %q\n\n", v.Message)
+	}
+	if v.Memo != "" {
+		if v.MemoType == "MEMO_TEXT" {
+			ui.Printf("Public memo: %q\n\n", v.Memo)
```

Comment:
Is this susceptible to Antonio-style trickery with tty control characters in the memo? Maybe that was remedied in ui.Printf, I don't remember.